pennylane/gradients/parameter_shift.py

```diff
@@ -61,27 +62,48 @@ def _square_observable(obs):
     return NONINVOLUTORY_OBS[obs.name](obs)
 
 
-def _get_operation_recipe(tape, t_idx, shift=np.pi / 2):
+def _get_operation_recipe(tape, t_idx, shifts):
     """Utility function to return the parameter-shift rule
     of the operation corresponding to trainable parameter
     t_idx on tape.
 
-    If the corresponding operation has grad_recipe=None, then
-    the default two-term parameter-shift rule is assumed.
+    This function performs multiple attempts to obtain the recipe:
+
+    - If ``parameter_frequencies`` yield a result, the frequencies are
+      used to construct the general parameter-shift rule via
+      ``qml.gradients.generate_shift_rule``
+    - If ``parameter_frequencies`` raises an error (because the operation
+      has no custom ``parameter_frequencies`` and no ``generator`` is defined),
+      the ``get_parameter_shift`` method of the operation is used.
+      If in turn no custom version of ``get_parameter_shift`` is defined
+      and the operation does not have a ``grad_recipe``, the two-term
+      parameter-shift rule is assumed.

```

Comment:
😬

Reply:
I don't think in-scope for this PR, but we should definitely remove this default behaviour

Reply:
Ticket created :)